File: 3773-minimum-pair-removal-to-sort-array-i/minimum-pair-removal-to-sort-array-i.java
import java.util.ArrayList;
import java.util.List;

class Solution {
    public int minimumPairRemoval(int[] nums) {
        List<Integer> list = new ArrayList<>();
        for (int num : nums) {
            list.add(num);
        }

        int operations = 0;
        
        while (true) {
            boolean isSorted = true;
            for (int i = 0; i < list.size() - 1; i++) {
                if (list.get(i) > list.get(i + 1)) {
                    isSorted = false;
                    break;
                }
            }
            
            if (isSorted) {
                return operations;
            }
            
            int minSum = Integer.MAX_VALUE;
            int mergeIndex = -1;
            
            for (int i = 0; i < list.size() - 1; i++) {
                int currentSum = list.get(i) + list.get(i + 1);
                if (currentSum < minSum) {
                    minSum = currentSum;
                    mergeIndex = i;
                }
            }
            
            list.set(mergeIndex, minSum);
            list.remove(mergeIndex + 1);
            
            operations++;
        }
    }
}

File: 3773-minimum-pair-removal-to-sort-array-i/minimum-pair-removal-to-sort-array-i.py
class Solution:
    def minimumPairRemoval(self, nums: List[int]) -> int:
        operations = 0
        while True:
            is_sorted = True
            for i in range(len(nums) - 1):
                if nums[i] > nums[i + 1]:
                    is_sorted = False
                    break
            
            if is_sorted:
                return operations
            
            min_sum = float('inf')
            merge_index = -1
            
            for i in range(len(nums) - 1):
                current_sum = nums[i] + nums[i + 1]
                if current_sum < min_sum:
                    min_sum = current_sum
                    merge_index = i
            
            nums[merge_index] = nums[merge_index] + nums[merge_index + 1]
            nums.pop(merge_index + 1)
            
            operations += 1